File: openframeworks/DeeSeventySix/src/Grain.cpp
#include "Grain.h"

Grain::Grain(float _x, float _y, float _energy, float _grainSize, int _minCrystals, int _maxCrystals) {
    x = _x;
    y = _y;
    minCrystals = _minCrystals;
    maxCrystals = _maxCrystals;
    energy = _energy;
    exposed = false;
    developed = false;
    grainSize = _grainSize;
    numCrystals = (int) ofRandom(minCrystals, maxCrystals);

    for (int i=0; i<numCrystals; i++) {
        crystals.push_back(Crystal(x + ofRandom(grainSize) - random(grainSize), y + random(grainSize) - random(grainSize)));
    }
}

File: DeeSeventySix_alifejs/js/grain.js
"use strict";

class Grain {    
    
    constructor(_x, _y, _energy, _grainSize, _minCrystals, _maxCrystals) {  // float, float, float, float, int, int
        this.x = _x;
        this.y = _y;
        this.minCrystals = _minCrystals;
        this.maxCrystals = _maxCrystals;
        this.energy = _energy;
        this.exposed = false;
        this.developed = false;
        this.grainSize = _grainSize;
        this.numCrystals = parseInt(random(this.minCrystals, this.maxCrystals));
        this.crystals = [];
        
        for (let i=0; i<this.numCrystals; i++) {
            this.crystals.push(new Crystal(this.x + random(-this.grainSize, this.grainSize), this.y + random(-this.grainSize, this.grainSize)));
        }
    }
    
}
